### ISSUE TITLE
Convert the OAV to ophyd-async 

### ISSUE THREAD

**Issue by DominicOram** [DiamondLightSource/dodal#716]
We should try and get all our devices into `ophyd-async`. The OAV is one of the last ones left

## Note
This is a reasonably large device, it may make sense to split it into multiple issues

## Acceptance Criteria
* The OAV is converted into `ophyd-async`
* Tests all still pass

**Comment by DominicOram** [comment]
https://github.com/DiamondLightSource/dodal/issues/801 should be done first

**Comment by DominicOram** [comment]
As part of this we should probably look at how we do the `OAVConfigParams`. Particularly how they get updated from the device. It may make more sense for them to be derived signals https://github.com/bluesky/ophyd-async/issues/525

**Comment by noemifrisina** [comment]
As this is a fairly large device and we probably can't have an ophyd-async device with ophyd components (or viceversa), the first step will be to make a new very basic ophyd-async device meeting the minimum needs for i24 serial and then we can add the rest of the functionality as we go along.
